Cognite.Extensions/CogniteResult.cs

```diff
@@ -211,6 +211,22 @@ public static CogniteResult<TError> Merge(params CogniteResult<TError>[]? result
             res.MergeErrors();
             return res;
         }
+
+        /// <summary>
+        /// Return a grouping of skipped error objects with a list of the errors that caused them to be skipped.
+        /// Useful for reporting a list of individual errors per passed resource.
+        /// </summary>
+        /// <returns>Errors grouped by skipped objects</returns>
+        public IEnumerable<(TError Skipped, IEnumerable<CogniteError<TError>> Errors)> ErrorsBySkipped()
+        {
+            if (Errors == null || !Errors.Any()) return Enumerable.Empty<(TError, IEnumerable<CogniteError<TError>>)>();
+
+            return Errors
+                .SelectMany(err => err.Skipped.Select(skipped => (skipped, err)))
+                .GroupBy(pair => pair.skipped)
```

Comment:
of what type is the skipped property here?  you group by `reason type` or `skipped/not skipped`? just curious

Reply:
Skipped is `TError`, so a generic type. I group by whatever the default equality comparer for that type is, which in most cases is going to be reference equality, which works well here. The idea is just to get the list of errors for each skipped element.